controllers/awscluster_controller.go

```diff
@@ -159,10 +196,10 @@ func (r *AWSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 func (r *AWSClusterReconciler) reconcileDelete(clusterScope *scope.ClusterScope) (reconcile.Result, error) {
```

Comment:
IMO the above checks should be done inside during `DeleteBastion()` & `DeleteSecurityGroups()` only, as we are not doing any such sanity checks in controllers.
You might want to add these comments for the pkg `securitygroups` and `bastion` in original issue so that it can be taken care of.